Fix issues with case sorting/searching from case dashboard

mattwire opened the pull request:
Overview
----------------------------------------
* Fixes https://lab.civicrm.org/dev/core/-/issues/1624: Case Dashboard/Dashlet gives datatables error if sort by type or status:
Now works!

* "Fixes" https://lab.civicrm.org/dev/core/-/issues/2319: Network error when trying to sort results by 'Next Scheduled date':
Disables sorting on most recent / next scheduled. This is better than before where it allows you to sort but then triggers a DB error whenever you try. It looks more complex to actually make the sort work because of the complexity of the query and dependencies internally in CiviCRM.

Before
----------------------------------------
500 / network error when sorting by status/type.

After
----------------------------------------
Sorting/searching works without error.

Technical Details
----------------------------------------
See lab issues

Comments
----------------------------------------
@demeritcowboy

mattwire committed:
Fix issues with case sorting/searching from case dashboard

demeritcowboy commented:
Thanks will check it out.

demeritcowboy commented:
I haven't looked closely yet but this has a weird effect on the Cases tab on a contact. It now shows two columns for Next Scheduled Date instead of one for Scheduled and one for Recent, and both are now blank.

demeritcowboy commented:
Maybe a good start but going to close this one until someone has time to make it work. The lab tickets link here already.

mattwire commented:
Hi @demeritcowboy I've just updated/refreshed this and updated the description above. I don't see any of the issues you mentioned introduced anymore and this should be an overall improvement in stuff actually working!

demeritcowboy commented:
Thanks will take a look.

mattwire commented:
Thanks @demeritcowboy
